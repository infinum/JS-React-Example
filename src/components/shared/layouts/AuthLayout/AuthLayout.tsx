import React, { FC } from 'react';
import { Box, BoxProps, Container } from '@chakra-ui/react';

export const AuthLayout: FC<BoxProps> = ({ children }) => {
	return (
		<Container size="md" py={64}>
			{children}
		</Container>
	);
};
