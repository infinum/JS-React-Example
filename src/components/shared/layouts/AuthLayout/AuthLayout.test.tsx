import { axe } from 'jest-axe';
import { render, screen } from 'test-utils';
import { AuthLayout } from './AuthLayout';

describe('AuthLayout', () => {
	it('should matches snapshot', () => {
		const { asFragment } = render(<AuthLayout />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should render children', () => {
		render(
			<AuthLayout>
				<p>test</p>
			</AuthLayout>
		);

		expect(screen.getByText('test')).toBeInTheDocument();
	});

	it('should be accessible', async () => {
		const { container } = render(<AuthLayout />);

		const results = await axe(container);

		expect(results).toHaveNoViolations();
	});
});
