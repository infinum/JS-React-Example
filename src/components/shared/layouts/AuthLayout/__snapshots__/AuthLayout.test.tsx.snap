// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLayout should matches snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: var(--chakra-sizes-full);
}

<div
    class="emotion-0"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
